### PR TITLE
feat: add json-parser.c, allowing for dealing with non-static json

### PR DESCRIPTION
**Pull request by LucasMull**

### CODE REVIEW

File: common/json-actor.c [hunk 1]
```diff
@@ -2421,25 +2421,19 @@ json_vextract (char * json, size_t size, char * extractor, va_list ap)
...
 
   /* Assume the top-level element is an object */
   if (!(tokens[0].type == JSMN_OBJECT || tokens[0].type == JSMN_ARRAY))
     ERR("Found %d, Object or array expected", tokens[0].type);
 
-  for (int i = 0; i < num_tok; i++) {
-    //print_tok(stderr, json, tokens, i);
-  }
-
   info.n_tokens = num_tok;
```
Comment (mlite) [inline common/json-actor.c:2437]: We need to keep this for debugging.

File: common/json-parser.c [hunk 2]
```diff
@@ -0,0 +1,1290 @@
...
+
+    return new_comp;
+}
+
+static char*
+_json_decode_string(char **p_buffer)
+{
+    char *start = *p_buffer;
+    ASSERT_S('\"' == *start, "Not a string");
+
```
Comment (mlite) [inline common/json-parser.c:193]: Can you please file an issue to replace this with jsmn's parser later?  It's better we only keep one parser in the code base to reduce the complexity and maintenance burden.

File: common/json-actor.c (hunk 1 above)
Comment (LucasMull) [reply common/json-actor.c:2437]: Sorry, I will add it back.

File: common/json-parser.c (hunk 2 above)
Comment (LucasMull) [reply common/json-parser.c:193]: Completely agree, I will file an issue.

Comment (LucasMull) [reply common/json-parser.c:193]: It will also help reduce a lot of code.

**Pull request merged**